Prose/ProseLib/Sources/App/AppReducer.swift

```diff
@@ -36,17 +36,15 @@ public let appReducer: Reducer<
             ))
 
         case let .auth(.loginResult(.success(jid))):
-            if let url = URL(string: "prose://main") {
-                return environment.openURL(url, .init())
-                    .receive(on: environment.mainQueue)
-                    .map { AppAction.didLogIn(jid: jid) }
-                    .mapError(EquatableError.init)
-                    .catch { Just(AppAction.showError($0)) }
-                    .eraseToEffect()
-            } else {
-                print("Error creating main window URL")
-                return Effect(value: .showError(EquatableError(NavigationError.cannotConstructURL)))
-            }
+            let url = URL(staticString: "prose://main")
+            return environment.openURL(url, .init())
+                .receive(on: environment.mainQueue)
+                .map { AppAction.didLogIn(jid: jid) }
+                .mapError(EquatableError.init)
+                .catch { error -> AnyPublisher<AppAction, Never> in
+                    fatalError("Failed to open URL <\(url.absoluteString)>: \(error.localizedDescription)")
+                }
+                .eraseToEffect()
 
         case let .showError(error):
```

Comment:
This is now unused.

Reply:
What do you mean?

Reply:
The `showError` enum case is unused. Most likely for other failable action the pattern would be to have an enum with a `Result` type where an alert property would be set when the action failed.

Reply:
Oh, good catch! I'm still not used to GitHub showing the selected line as the last line… I thought you were talking about `.catch`